src/bin/cargo/cli.rs

```diff
@@ -473,6 +480,14 @@ See 'cargo help <command>' for more information on a specific command.\n",
                 .value_name("WHEN")
                 .global(true),
         )
+        .arg(
+            opt("directory", "Change to DIRECTORY before doing anything")
```

Comment:
Could you add a test for this?

Reply:
I added two pretty minimal tests.  Moving additional discussion over to there.